flink-runtime/src/test/java/org/apache/flink/runtime/metrics/MetricReporterSetupTestBase.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.metrics;
+
+import org.apache.flink.metrics.reporter.MetricReporterFactory;
+import org.apache.flink.util.TestLogger;
+
+import org.junit.Test;
+
+import java.util.Iterator;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * The base class to verify metric reporter factory.
+ */
+public abstract class MetricReporterSetupTestBase extends TestLogger {
+
+	@Test
+	public void testMetricReporterSetupViaSPI() {
+		Iterator<MetricReporterFactory> factoryIterator = ReporterSetup.getAllReporterFactoriesViaSPI();
```

Comment:
It would be good to get rid of the dependency to flink-runtime; this change goes contrary to our goals of separating the metric modules from flink-runtime.
The ServiceLoader mechanism is standardized, so it's fine to just call `ServiceLoader.load(MetricReporterFactory.class).iterator()` here.